Tests/AsyncHTTPClientTests/HTTPClientTests.swift

```diff
@@ -2008,6 +2008,40 @@ class HTTPClientTests: XCTestCase {
         self.defaultClient = nil // so it doesn't get shut down again.
     }
 
+    func testConnectErrorPropagatedToDelegate() throws {
+        class TestDelegate: HTTPClientResponseDelegate {
+            typealias Response = Void
+            var error: Error?
+            func didFinishRequest(task: HTTPClient.Task<Void>) throws {}
+            func didReceiveError(task: HTTPClient.Task<Response>, _ error: Error) {
+                self.error = error
+            }
+        }
+
+        let httpBin = HTTPBin()
+        let httpClient = HTTPClient(eventLoopGroupProvider: .shared(self.clientGroup))
+        defer {
+            XCTAssertNoThrow(try httpClient.syncShutdown())
+        }
+
+        let delegate = TestDelegate()
+        let request = try HTTPClient.Request(url: "http://localhost:\(httpBin.port)/get")
+        do {
+            try httpBin.shutdown()
+            _ = try httpClient.execute(request: request, delegate: delegate).wait()
```

Comment:
please use

```
XCTThrowsError(try httpClient.execute(...).wait()) { error in
   XCTAssert(...)
}
```

we literally had security issues before because of this pattern.

Reply:
done, thank you for catching this!